test/UniswapV3Staker.unit.spec.ts

```diff
@@ -852,6 +868,118 @@ describe('UniswapV3Staker.unit', async () => {
     })
   })
 
+  describe('#claimReward', () => {
+    let rewardToken: TestERC20
+    let startTime: number
+    let endTime: number
+    let claimDeadline: number
+    let tokenId: string
+
+    beforeEach('setup', async () => {
+      const currentTime = await blockTimestamp()
+      rewardToken = tokens[2]
+      startTime = currentTime
+      endTime = currentTime + 100
+      claimDeadline = currentTime + 200
+
+      await tokens[0].connect(wallets[0]).transfer(lpUser0.address, 100)
+      await tokens[1].connect(wallets[0]).transfer(lpUser0.address, 100)
+
+      await tokens[0].connect(lpUser0).approve(nft.address, 100)
+      await tokens[1].connect(lpUser0).approve(nft.address, 100)
+
+      await rewardToken
+        .connect(wallets[0])
+        .transfer(incentiveCreator.address, totalReward)
+
+      await rewardToken
+        .connect(incentiveCreator)
```

Comment:
I personally think assigning roles to every action in these unit tests goes above and beyond the call of duty and potentially overcomplicates things...but I know omar likes it so.... 🤷 😂 I'm not going to consider it a blocker